QA system bug: what is the capital of canada

Here's what I'm getting:

```
vpn-019:js jimmylin$ curl -H "Content-Type: application/json" -X POST -d '{"question": "what is the capital of canada"}' http://localhost:5546/answer
{
  "error": "ERROR - could not parse the question or get answer."
}
```

@salman1993 @rosequ any ideas?

This was the error:

```
Question: what is the capital of Canada
An error occurred while calling o10.getRankedPassages.
: org.apache.lucene.queryparser.classic.ParseException: Cannot parse 'Shell Canada Sets Revamp With Cuts In Jobs, Big Charge ---- By Gary Lamphier Staff Reporter of The Wall Street Journal 06/01/90 CALGARY, Alberta -- Shell Canada Ltd., facing shrinking earnings and hefty spending plans, said it will restructure its oil refining, marketing and chemical businesses, which last year generated 87% of the company's 4.92 billion Canadian dollars (US$4.17 billion) in revenue.': Encountered " "-" "- "" at line 1, column 56.
Was expecting one of:
    <BAREOPER> ...
    "(" ...
    "*" ...
    <QUOTED> ...
    <TERM> ...
    <PREFIXTERM> ...
    <WILDTERM> ...
    <REGEXPTERM> ...
    "[" ...
    "{" ...
    <NUMBER> ...
    <TERM> ...
    "*" ...

	at org.apache.lucene.queryparser.classic.QueryParserBase.parse(QueryParserBase.java:116)
	at io.anserini.qa.passage.IdfPassageScorer.score(IdfPassageScorer.java:56)
	at io.anserini.py4j.PyseriniEntryPoint.getRankedPassages(PyseriniEntryPoint.java:141)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:280)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.apache.lucene.queryparser.classic.ParseException: Encountered " "-" "- "" at line 1, column 56.
```

--- a/src/main/java/io/anserini/py4j/PyseriniEntryPoint.java
+++ b/src/main/java/io/anserini/py4j/PyseriniEntryPoint.java
@@ -130,6 +130,7 @@ public class PyseriniEntryPoint {
   public List<String> getRankedPassages(String query, int numHits, int k) throws Exception {
     List<String> docids = search(query, numHits);
     List<String> sentencesList = new ArrayList<>();
+    sentencesList.add(query);
     for (String docid : docids) {
       List<Sentence> sentences = indexUtils.getSentDocument(docid);
       for (Sentence sent : sentences) {

--- a/src/main/python/api.py
+++ b/src/main/python/api.py
@@ -57,4 +57,4 @@ if __name__ == "__main__":
     print("Port: {}".format(app.config['port']))
     print("Debug info: {}".format(args.debug))
 
-    app.run(debug=args.debug, host=app.config['host'], port=app.config['port'])
+    app.run(debug=args.debug, host=app.config['host'], port=int(app.config['port']))

